infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -497,7 +514,12 @@ public void updateSiteConfig(LogReplicationClusterInfo.TopologyConfigurationMsg
      */
     @Override
     public void prepareSiteRoleChange() {
-        replicationManager.prepareSiteRoleChange();
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && replicationManager != null) {
+            replicationManager.prepareSiteRoleChange();
```

Comment:
same here... in ReplicationManager let's rename for consistency to prepareClusterRoleChange